daemon/attach.go

```diff
@@ -3,33 +3,30 @@ package daemon
 import (
 	"fmt"
 	"io"
-	"net/http"
 	"time"
 
 	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/container"
 	"github.com/docker/docker/daemon/logger"
 	derr "github.com/docker/docker/errors"
 	"github.com/docker/docker/pkg/stdcopy"
 )
 
-// ContainerAttachWithLogsConfig holds the streams to use when connecting to a container to view logs.
-type ContainerAttachWithLogsConfig struct {
-	Hijacker   http.Hijacker
-	Upgrade    bool
+// ContainerAttachConfig holds the streams to use when connecting to a container to view logs.
+type ContainerAttachConfig struct {
+	Streamer   types.Streamer
 	UseStdin   bool
 	UseStdout  bool
 	UseStderr  bool
 	Logs       bool
 	Stream     bool
 	DetachKeys []byte
+	Mux        bool
```

Comment:
:ghost: It is a mystery. :ghost: 

I did think it was funny it just passes the same stream three times with different names in the config (for websockets). But yes, that is essentially what does in both sides is hijack the stream and multiplex all communication through one single stream. There is a further refactor that might be able to be done to unify everything, but that is a big change to the API.